x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/indexlifecycle/LifecycleAction.java

```diff
@@ -19,7 +19,12 @@
 package org.elasticsearch.protocol.xpack.indexlifecycle;
 
 /**
- * Marker interface for index lifecycle management actions
+ * interface for index lifecycle management actions
  */
-public class LifecycleAction {
+public interface LifecycleAction {
+
+    /**
+     * @return the name of this action
+     */
+    String getName();
```

Comment:
nit: could we call this `getType()` instead? Elsewhere in the codebase we often use `name` for user specified things and `type` for types of implementation where the name is fixed. `NamedWriteable` breaks that a bit for me which bothers me but I think thats the only major exception and isn't user facing

Reply:
I'm not sure `type` is the more appropriate name here. The only reason this method exists is for  the ability for Phase to leverage its name when looking up NamedXContent. Since this would further diverge the terminology away from our Named- naming world, I feel like calling it `type` here would fix much. Also, we have these actions being supported by specific LifecycleTypes (yet another `type`), that I think there is further confusion to be had there.

Reply:
Fair enough

Reply:
that being said, I agree that it is unfortunate there is no strong consistency here. For posterity, I'll link to the equivalent info from the Ingest world which uses `getType`: https://github.com/elastic/elasticsearch/blob/cf7489899a957764404769a4b792cdbefbf6a729/modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/GrokProcessor.java#L86-L89